examples/hal/compute/main.rs

```diff
@@ -0,0 +1,162 @@
+        let mut writer = gpu.device.acquire_mapping_writer::<u32>(&staging_buffer, 0..stride * numbers.len() as u64).unwrap();
+        writer.copy_from_slice(&numbers);
+        gpu.device.release_mapping_writer(writer);
+    }
+
+    let (device_memory, device_buffer) = create_buffer(
+        &mut gpu,
+        memory::DEVICE_LOCAL,
+        buffer::TRANSFER_DST,
+        stride,
+        numbers.len() as u64
+    );
+
+    let desc_set = desc_pool.allocate_sets(&[&set_layout]).remove(0);
+    gpu.device.update_descriptor_sets(&[
+        pso::DescriptorSetWrite {
+            set: &desc_set,
+            binding: 0,
+            array_offset: 0,
+            write: pso::DescriptorWrite::StorageBuffer(vec![(&device_buffer, 0..stride * numbers.len() as u64)])
+        }
+    ]);
+
+    let mut command_pool = gpu.compute_queues[0].create_compute_pool(16, pool::CommandPoolCreateFlags::empty());
+    let fence = gpu.device.create_fence(false);
+    let submission = queue::Submission::new().submit(&[{
+        let mut command_buffer = command_pool.acquire_command_buffer();
+        command_buffer.copy_buffer(&staging_buffer, &device_buffer, &[command::BufferCopy { src: 0, dst: 0, size: stride * numbers.len() as u64}]);
+        command_buffer.bind_compute_pipeline(&pipeline);
+        command_buffer.bind_compute_descriptor_sets(&pipeline_layout, 0, &[&desc_set]);
+        command_buffer.dispatch(numbers.len() as u32, 1, 1);
+        command_buffer.copy_buffer(&device_buffer, &staging_buffer, &[command::BufferCopy { src: 0, dst: 0, size: stride * numbers.len() as u64}]);
+        command_buffer.finish()
+    }]);
+    gpu.compute_queues[0].submit(submission, Some(&fence));
+    gpu.device.wait_for_fences(&[&fence], device::WaitFor::All, !0);
+
+    {
+        let reader = gpu.device.acquire_mapping_reader::<u32>(&staging_buffer, 0..stride * numbers.len() as u64).unwrap();
+        println!("Times: {:?}", reader.into_iter().map(|n| *n).collect::<Vec<u32>>());
+        gpu.device.release_mapping_reader(reader);
+    }
+
+    gpu.device.destroy_descriptor_pool(desc_pool);
+    gpu.device.destroy_descriptor_set_layout(set_layout);
+    gpu.device.destroy_shader_module(shader);
+    gpu.device.destroy_buffer(device_buffer);
+    gpu.device.destroy_buffer(staging_buffer);
+    gpu.device.destroy_fence(fence);
+    gpu.device.destroy_pipeline_layout(pipeline_layout);
+    gpu.device.free_memory(device_memory);
+    gpu.device.free_memory(staging_memory);
+    gpu.device.destroy_compute_pipeline(pipeline);
+}
+
+fn create_buffer<B: Backend>(gpu: &mut Gpu<B>, properties: memory::Properties, usage: buffer::Usage, stride: u64, len: u64) -> (B::Memory, B::Buffer) {
+    let ty = (&gpu.memory_types).into_iter().find(|memory_type| {
+        memory_type.properties.contains(properties)
+    }).unwrap();
+
+    let memory = gpu.device.allocate_memory(&ty, stride * len).unwrap();
+    let buffer = gpu.device.create_buffer(stride * len, stride, usage).unwrap();
```

Comment:
The created buffer has certain memory requirements, it might be larger then the size specified and only supports specific heaps. A more robust implementation would be for this example to:
- `create_buffer`
- request memory requirements
- `allocate_memory` with values from the requirements
- `bind_buffer_memory`